### ISSUE TITLE
CA1008: Allow other names for 0-value member of [Flags] enums

### ISSUE THREAD

**Issue by Rabadash8820** [dotnet/roslyn-analyzers#5777]
### Analyzer

**Diagnostic ID**: [CA1008](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1008)

### Describe the improvement

In addition to `None`, the 0-value member of `[Flags]` `enum`s should be allowed to have one of these names:

- `Never`
- `Nothing`

### Describe suggestions on how to achieve the rule

Sorry, I'm not sure what this is asking...

### Additional context

I came across this warning when I tried to declare the following `enum`:

```cs
    [Flags]
    public enum ConfigurationLoadContext {
        Never = 0b0000,    // CA1008	In enum ConfigurationLoadContext, change the name of Never to 'None'
        BuildScript = 0b0001,
        PlayMode = 0b0010,
        DebugBuild = 0b0100,
        ReleaseBuild = 0b1000,
        Always = 0b1111,
    }
```

Now, maybe I'm being stubborn, but I think `Never` is just as descriptive as `None` in this context. In fact, I would say that any English word that conveys a "lack of all the things" is valid for the 0-member of a `[Flags]` `enum`. The only single words that I could think of with this connotation were:
- `Never`: suggested above
- `Nothing`: suggested above
- `Zero`: connotes the actual number, not the concept of "nothing", so I'm not suggesting it...but I could be convinced otherwise
- `Notta` or `Nada`: not real English words ([I think?](https://www.urbandictionary.com/define.php?term=notta)), so I'm not suggesting them
- `Naught`, `Nix`, or `Nil`: while these _are_ real English words (according to Scrabble and Merriam-Webster), I highly doubt that anyone would actually use them in place of the more common words above

**Comment by Evangelink** [comment]
@mavasani I think the easiest would be to add a user configuration to provide a list of alternate names.

It would still be worth reviewing the suggestions to see if we want to "hardcode" some of them.

**Comment by mavasani** [comment]
> I think the easiest would be to add a user configuration to provide a list of alternate names.

Sounds good to me, its worth adding an option here.

> It would still be worth reviewing the suggestions to see if we want to "hardcode" some of them.

If we get more feedback to include more hardcoded names by default, we can do so later.
